Riot/Modules/VoiceBroadcast/VoiceBroadcastSDK/VoiceBroadcastService.swift

```diff
@@ -0,0 +1,295 @@
+//
+// Copyright 2022 New Vector Ltd
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import Foundation
+
+/// Voice Broadcast settings.
+@objcMembers
+final class VoiceBroadcastSettings: NSObject {
+    static let eventType = "io.element.voice_broadcast_info"
+    
+    static let voiceBroadcastContentKeyState = "state"
+    static let voiceBroadcastContentKeyChunkLength = "chunk_length"
+}
+
+/// VoiceBroadcastService handles voice broadcast.
+/// Note: Cannot use a protocol because of Objective-C compatibility
+@objcMembers
+public class VoiceBroadcastService: NSObject {
+    
+    // MARK: - Properties
+    
+    private var voiceBroadcastInfoEventId: String?
+    public let room: MXRoom
+    public private(set) var state: State
+    
+    // MARK: - Setup
+    
+    public init(room: MXRoom, state: State) {
+        self.room = room
+        self.state = state
+    }
+
+    // MARK: - Constants
+    
+    public enum State: String {
+        case started
+        case paused
+        case resumed
+        case stopped
+    }
+    
+    // MARK: - Public
+    
+    // MARK: Voice broadcast info
+        
+    /// Start a voice broadcast.
+    /// - Parameters:
+    ///   - completion: A closure called when the operation completes. Provides the event id of the event generated on the home server on success.
+    /// - Returns: a `MXHTTPOperation` instance.
+    func startVoiceBroadcast(completion: @escaping (MXResponse<String?>) -> Void) -> MXHTTPOperation? {
+        return sendVoiceBroadcastInfo(state: State.started) { (response) in
```

Comment:
Optional as you have more context: Wondering if it be sensible to add a `[weak self]` to this closure (and a few others) given `self` is used?